### PR TITLE
Adding windows system_info virtual table

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by facebook-github-bot-6** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2521/files/78ad1e1818ccf6c226f2f3b4e6cfe87efc680354..a0f85c27ed37f8b4c02e1e232b0d74803134d764)

**Comment by osqueryer** [comment]
:no_good: The commit a0f85c27ed37f8b4c02e1e232b0d74803134d764 (Job results: [337](https://jenkins.osquery.io/job/osqueryPullRequestAudit/337/)) failed the code audit.

**Comment by facebook-github-bot-3** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2521/files/a0f85c27ed37f8b4c02e1e232b0d74803134d764..fd1009be62636aa7d099eef4a20aaf02b91ce3bb)

File: osquery/tables/system/windows/system_info.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  auto qd = SQL::selectAllFrom("cpuid");
+  for (const auto& row : qd) {
+    if (row.at("feature") == "product_name") {
+      r["cpu_brand"] = row.at("value");
+      boost::trim(r["cpu_brand"]);
+    }
+  }
+
+  WmiRequest wmiUUIDReq("Select UUID from Win32_ComputerSystemProduct");
```
Comment (theopolis) [inline osquery/tables/system/windows/system_info.cpp]: Do you mind adding this to the `generateUUID` method in `system.cpp` with a new `#elif` to differentiate Windows from Linux?

Comment (theopolis) [inline osquery/tables/system/windows/system_info.cpp]: That is, after #2522 is merged.

Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/system_info.cpp]: 👍

**Comment by theopolis** [comment]
Good to go on the rebase now.

**Comment by facebook-github-bot-4** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2521/files/fd1009be62636aa7d099eef4a20aaf02b91ce3bb..c27a0610eb8f09379496ef7bbf6b117756928897)

**Comment by osqueryer** [comment]
:no_good: The commit c27a0610eb8f09379496ef7bbf6b117756928897 (Job results: [342](https://jenkins.osquery.io/job/osqueryPullRequestAudit/342/)) failed the code audit.

**Comment by facebook-github-bot-2** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2521/files/c27a0610eb8f09379496ef7bbf6b117756928897..240f6b8019aa96e82467b6861b233f17de7015c5)

File: osquery/CMakeLists.txt [hunk 2]
```diff
@@ -304,6 +304,10 @@ if(NOT DEFINED ENV{SKIP_TESTS})
   add_executable(osquery_tests main/tests.cpp ${OSQUERY_TESTS})
   ADD_DEFAULT_LINKS(osquery_tests FALSE)
   target_link_libraries(osquery_tests gtest gmock libosquery_testing)
+  if(WINDOWS)
+    # Linked for the WMI libs
+    target_link_libraries(osquery_tests Release/osquery_additional)
```
Comment (theopolis) [inline osquery/CMakeLists.txt]: This isn't a good workaround. To make this work with the build the WMI classes will need to be available to core. Mind opening an issue for this need to refactor?

Comment (PoppySeedPlehzr) [inline osquery/CMakeLists.txt]: Done. If this poses a threat I can also try and get the refactor cranked out with this PR? https://github.com/facebook/osquery/issues/2524

**Comment by theopolis** [comment]
The WMI code could live in /core/windows for now.

**Comment by facebook-github-bot-1** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2521/files/240f6b8019aa96e82467b6861b233f17de7015c5..f681409062279e83b18510eb70666a7ad0cccde1)

**Comment by osqueryer** [comment]
:no_good: The commit f681409062279e83b18510eb70666a7ad0cccde1 (Job results: [344](https://jenkins.osquery.io/job/osqueryPullRequestAudit/344/)) failed the code audit.

**Comment by facebook-github-bot-5** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2521/files/f681409062279e83b18510eb70666a7ad0cccde1..c3199e68c158d83fd551913af492452b9db4cb29)

**Pull request merged**